<?xml version="1.0"?>
<!-- SPDX-FileCopyrightText: 2020 Smart Robotics Lab, Imperial College London -->
<!-- SPDX-FileCopyrightText: 2020 Sotiris Papatheodorou -->
<!-- SPDX-License-Identifier: CC0-1.0 -->
<launch>
  <!-- Set the ROS log level in rosconsole.conf -->
  <env name="ROSCONSOLE_CONFIG_FILE" value="$(find supereight_ros)/config/rosconsole.conf"/>

  <!-- habitat-ros node and MAV simulator -->
  <remap from="/supereight/path" to="/mav_sim/goal_path"/>
  <include file="$(find habitat_ros)/launch/habitat_ros_sim.launch">
    <arg name="habitat_ros_config_file"         value="$(find supereight_ros)/config/habitat.yaml"/>
    <arg name="habitat_ros_mav_sim_config_file" value="$(find supereight_ros)/config/habitat_sim.yaml"/>
  </include>

  <!-- supereight-ros node -->
  <node pkg="supereight_ros"
        name="semanticeight"
        type="supereight_ros"
        clear_params="true"
        output="screen"
        required="true"
        launch-prefix="">
    <!-- Include rosparam yaml files -->
    <rosparam command="load" file="$(find supereight_ros)/config/semanticeight.yaml"/>
    <rosparam command="load" file="$(find supereight_ros)/config/habitat.yaml"/>
    <!-- Remap topic names -->
    <remap from="/camera/depth_image" to="/habitat/depth/image_raw"/>
    <remap from="/camera/rgb_image"   to="/habitat/rgb/image_raw"/>
    <remap from="/pose"               to="/habitat/pose"/>
    <remap from="/camera/class"       to="/habitat/semantic_class/image_raw"/>
    <remap from="/camera/instance"    to="/habitat/semantic_instance/image_raw"/>
  </node>

  <!-- rviz with appropriate config -->
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find supereight_ros)/config/semanticeight.rviz" />
</launch>

